model/riscv_insts_vext_arith.sail

```diff
@@ -155,245 +155,270 @@ mapping vvtype_mnemonic : vvfunct6 <-> string = {
+                      },
+        NV_VNCLIP  => {
+                        let v_double : bits('m * 4) = EXTS(vs2_val[i]);
+                        let result_wide = slice(v_double >> shift_amount, 0, 'o) + EXTZ('o, rounding_incr);
+                        signed_saturation('m, result_wide);
+                      }
       }
     }
   };
 
   write_vreg(num_elem, SEW, LMUL_pow, vd, result);
   vstart = EXTZ(0b0);
   RETIRE_SUCCESS
-} 
+}
 
-mapping visg_mnemonic : visgfunct6 <-> string = {
-  VI_VSLIDEUP     <-> "vslideup.vi",
-  VI_VSLIDEDOWN   <-> "vslidedown.vi",
-  VI_VRGATHER     <-> "vrgather.vi"
+mapping nvtype_mnemonic : nvfunct6 <-> string = {
+  NV_VNCLIPU     <-> "vnclipu.wv",
+  NV_VNCLIP      <-> "vnclip.wv"
 }
 
-mapping clause assembly = VISG(funct6, vm, vs2, simm, vd)
-  <-> visg_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ reg_name(simm) ^ maybe_vmask(vm)
+mapping clause assembly = NVTYPE(funct6, vm, vs2, vs1, vd)
+  <-> nvtype_mnemonic(funct6) ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs2) ^ sep() ^ vreg_name(vs1) ^ maybe_vmask(vm)
 
-/* ********************* Whole Vector Register Move (OPIVI) ********************** */
-union clause ast = VMVRTYPE : (regidx, bits(5), regidx)
+/* ********************** OPIVV (Integer Merge Instruction) ********************** */
+union clause ast = MASKTYPEV : (regidx, regidx, regidx)
 
-mapping clause encdec = VMVRTYPE(vs2, simm, vd) if haveRVV()
-  <-> 0b100111 @ 0b1 @ vs2 @ simm @ 0b011 @ vd @ 0b1010111 if haveRVV()
+mapping clause encdec = MASKTYPEV (vs2, vs1,  vd) if haveRVV()
+  <-> 0b010111 @ 0b0 @ vs2 @ vs1 @ 0b000 @ vd @ 0b1010111 if haveRVV()
 
-function clause execute(VMVRTYPE(vs2, simm, vd)) = {
-  let SEW     = get_sew();
-  let imm_val = unsigned(EXTZ(sizeof(xlen), simm));
-  let EMUL    = imm_val + 1;
+function clause execute(MASKTYPEV(vs2, vs1, vd)) = {
+  let start_element = get_start_element();
+  let end_element   = get_end_element();
+  let SEW           = get_sew();
+  let LMUL_pow      = get_lmul_pow();
+  let num_elem      = get_num_elem(LMUL_pow, SEW);
 
-  assert(EMUL == 1 | EMUL == 2 | EMUL == 4 | EMUL == 8);
-  let EMUL_pow = log2(EMUL);
-  let num_elem = get_num_elem(EMUL_pow, SEW);
   let 'n = num_elem;
   let 'm = SEW;
 
-  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, EMUL_pow, vs2);
-  write_vreg(num_elem, SEW, EMUL_pow, vd, vs2_val);
-  vstart = EXTZ(0b0);
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, 0b0, vreg_name("v0"));
+  let vs1_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs1);
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+  result      : vector('n, dec, bits('m)) = undefined;
 
-  RETIRE_SUCCESS
-} 
+  let tail_ag : agtype = get_vtype_vta();
+  foreach (i from 0 to (num_elem - 1)) {
+    if i < start_element then {
```

Comment:
Why is vstart supported in vmerge instructions but not in other arithmetic operations? Seems arbitrary as there is zero reason why vmerge could be interruptable in the middle of an operation when arithmetic operations aren't. The spec is quite flexible in the support for vstart in non memory operations so hardware implementations will have ample choice. In general interrupting a LMUL <=1 non-memory operation is probably not supported so vstart may only supported at destination register limits (LMUL > 1 at every step). So you can define in the SAIL model that vmerge can and other not, it's legal, but it's unlikely to match a given hardware implementation and you may want to specify why you made this choice.